File: app/src/main/java/cosmic/com/pkwprj/model/DbHelper.java
package cosmic.com.pkwprj.model;

import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

import androidx.annotation.Nullable;

import java.util.ArrayList;

public class DbHelper extends SQLiteOpenHelper {
    final static String TAG = "DBHelper";

    public DbHelper(@Nullable Context context, @Nullable String name, @Nullable SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
    }


    @Override
    public void onCreate(SQLiteDatabase db) {
        Log.d(TAG, "데이터 온크레이트호출");
        db.execSQL("CREATE TABLE HUB (_id INTEGER PRIMARY KEY AUTOINCREMENT, login TEXT, avatar_url TEXT, html_url TEXT, score INTEGER);");
//        db.execSQL( "CREATE TABLE GITHUBUSER (_id INTEGER PRIMARY KEY AUTOINCREMENT, user TEXT, url TEXT);" );
        Log.d(TAG, "db 생성");
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }

    public void insert(String name, String url, String html, int etc) {
        SQLiteDatabase db = getWritableDatabase();
        db.execSQL("INSERT INTO HUB VALUES(null, '" + name + "', '" + url + "', '" + html + "', '" + etc + "');");
        Log.d("TAG", "데이터 인서트확인");
    }

    public String getData(String name){
        SQLiteDatabase db = getReadableDatabase();
        String result = "";

        try {
            Log.d(TAG,"체크아이디");
            Cursor cursor=db.rawQuery( "SELECT * FROM HUB WHERE login='" + name + "';",null);
            while(cursor.moveToNext()){
                result =cursor.getString(1);
            }
        }catch (Exception e){

        }

        db.close();
        return result;
    }



    public void delete(String name) {
        SQLiteDatabase db = getWritableDatabase();
        db.execSQL("DELETE FROM HUB WHERE login='" + name + "';");
        db.close();
        Log.d(TAG, "데이터삭제");
    }

    public ArrayList<GithubOwner> getDataList() {

//    public String getDataList(){
        ArrayList<GithubOwner> dataList = new ArrayList<>();

//        String result=null;
        SQLiteDatabase db = getReadableDatabase();
        Cursor cursor = db.rawQuery("SELECT * FROM HUB", null);
        //
        if (cursor.moveToNext()) {
            do {
                GithubOwner userData = new GithubOwner();
                userData.setLogin( cursor.getString( 1 ) );
                userData.setAvatar_url(cursor.getString(2));
                userData.setHtml_url(cursor.getString(3));
                userData.setScore(cursor.getInt(4));

                dataList.add( userData);
            }
            while (cursor.moveToNext());

        }
        db.close();

        return dataList;
    }
}

File: app/src/main/java/cosmic/com/pkwprj/model/DbHelper.kt
package cosmic.com.pkwprj.model

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import java.util.*

class DbHelper(
    context: Context?,
    name: String?,
    factory: SQLiteDatabase.CursorFactory?,
    version: Int
) : SQLiteOpenHelper(context, name, factory, version) {

    //
    val dataList: ArrayList<GithubOwner>
        get() {

            val dataList = ArrayList<GithubOwner>()

            val db = readableDatabase
            val cursor = db.rawQuery("SELECT * FROM HUB", null)
            if (cursor.moveToNext()) {
                do {
                    val userData = GithubOwner()
                    userData.login=cursor.getString(1)
                    userData.avatar_url = cursor.getString(2)
                    userData.html_url = cursor.getString(3)
                    userData.score = cursor.getFloat(4)


                    dataList.add(userData)
                } while (cursor.moveToNext())

            }
            db.close()

            return dataList
        }


    override fun onCreate(db: SQLiteDatabase) {

        db.execSQL("CREATE TABLE HUB (_id INTEGER PRIMARY KEY AUTOINCREMENT, login TEXT, avatar_url TEXT, html_url TEXT, score FLOAT);")
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {

    }

    fun insert(name: String, url: String, html: String, etc: Float) {

        val db = writableDatabase
        db.execSQL("INSERT INTO HUB VALUES(null, '$name', '$url', '$html', '$etc');")
    }

    fun getData(name: String): String {
        val db = readableDatabase
        var result = ""

        try {
            val cursor = db.rawQuery("SELECT login FROM HUB WHERE login = '$name';", null)
            while (cursor.moveToNext()) {
                result = cursor.getString(0)
            }
        } catch (e: Exception) {

        }

        db.close()

        return result
    }


    fun delete(name: String) {
        val db = writableDatabase
        db.execSQL("DELETE FROM HUB WHERE login='$name';")
        db.close()
    }

    companion object {
        internal val TAG = "DBHelper"
    }
}
